packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1836,16 +1832,14 @@ export class KeyringController extends BaseController<
   }
 
   /**
-   * Create keyring.
+   * Create a new keyring, ensuring that the first account is
+   * also created.
    *
-   * - Creates a new vault.
-   * - Creates a new keyring with at least one account.
-   * - Makes the first account the selected account.
    * @param type - Keyring type to instantiate.
    * @param opts - Optional parameters required to instantiate the keyring.
-   * @returns A promise that resolves if the operation was successful.
+   * @returns A promise that resolves if the operation is successful.
    */
-  async #createKeyring(type: string, opts?: unknown) {
+  async #createKeyringWithFirstAccount(type: string, opts?: unknown) {
     const keyring = (await this.addNewKeyring(type, opts)) as EthKeyring<Json>;
 
     const [firstAccount] = await keyring.getAccounts();
```

Comment:
This method has been renamed to better match what it is used for

Reply:
I'm now realizing this naming might also be incorrect since the first account is not explicitly created here, but only _expected_ to be created - and this is a case only for HD keyring because we explicitly check for that in the `addNewAccount` method.

I think there might be a better way to do all this (see also my previous comment: https://github.com/MetaMask/core/pull/3830#discussion_r1466329733)